### ISSUE TITLE
508-defect-2: Contestable issue needs focus management 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#54645]
## Point of contact

joshkimux

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

In the following video you'll notice the following issues with focus management:
1. Focus returns to the `h3` title when _cancelling_ an addition
2. Focus returns to the `h3` title when _cancelling_ an edit
3. Focus returns to the prior issue when removing an issue

https://user-images.githubusercontent.com/14154792/223233903-ce5ed6a5-dcd6-41d3-946f-2c68e06efc4b.mov

## Reproduction steps

1. Cancel an addition, confirm focus goes to the `h3`
2. Cancel an edit, confirm focus goes to the `h3`
3. Remove an issue, confirm focus returns to the prior issue

## Proposed solution or next steps

- [x] Focus should return to "Add a new issue" upon removing an issue
- [ ] Focus should return to "Add a new issue" upon cancelling adding a new issue
- [ ] Focus should return to "Edit" when cancelling an edit action (no changes were made, so focus should return back to the link that triggered it)

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
> Focus returns to the h3 title when cancelling an edit

In the work that Rocio just completed, we returned focus to the edit button. Did you want to change this to the h3 instead?

**Comment by saderagsdale** [comment]
@joshkimux can you look at this one too?

**Comment by joshkimux** [comment]
https://user-images.githubusercontent.com/14154792/225377267-f6409741-074d-430d-912d-1e7f8c28332e.mov

Silky smooth, Robin carrying on his 🐐 legacy
